__tests__/src/config.test.ts

```diff
@@ -90,6 +90,18 @@ describe("Given a configuration", () => {
 
   })
 
+  it("When URLs contains '#', encodes it correctly", () => {
+    process.env.GITHUB_SHA = "123abc#ert516"
+    process.env.GITHUB_REPOSITORY = "vmware/vib-action"
+
+    const config = configFactory.getConfiguration()
+
+    if (config.shaArchive && config.shaArchive.includes("#")) {
+      const encodedShaArchive = encodeURIComponent(config.shaArchive)
+      expect(encodedShaArchive).not.toContain("#")
```

Comment:
Maybe it would be good to have an additional `expect` to make sure `encodeURIComponent` is returning the `%23` character?